cli/src/main/java/com/devonfw/tools/ide/npm/NpmJsonDist.java

```diff
@@ -0,0 +1,20 @@
+package com.devonfw.tools.ide.npm;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * JSON data object for a version of Npm. We map only properties that we are interested in and let jackson ignore all others.
+ *
+ * @see NpmJsonObject#getVersions()
+ */
+public class NpmJsonDist {
```

Comment:
Records are immutable and existing frameworks like hibernate, jackson, mapstruct, etc. are only slowly adding support for that since it is way complicated to unmarshall to a record and adding such support requires Java support for records what again requires "Java >= 14" and drops support for older Java versions.
While we might think that Java 17 or 21 should already be state of the art everywhere, there are still projects using Java8 in the wild and some of these libraries do not want to leave such projects left in the rain...
However, for jackson it seems that this should already work:
https://stackoverflow.com/questions/68394911/why-record-class-cant-be-properly-deserialized-with-jackson
So we could go for this even though surely a nice-to-have change...

Reply:
I have changed almost all of the classes that represent JSON data objects to be records.